File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> chars;
        int left = 0;
        int maxLength = 0;
        
        for (int right = 0; right < s.size(); right++) {
            while (chars.count(s[right])) {
                chars.erase(s[left]);
                left++;
            }

            chars.insert(s[right]);

            maxLength = max(maxLength, right - left + 1);
        }

        return maxLength;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers) - 1

        while left < right:
            curr = numbers[left] + numbers[right]
            if curr > target:
                right -= 1
            elif curr < target:
                left += 1
            elif curr == target:
                return [left + 1, right + 1]
            